File: PHP/car.php
<?php
require_once('account.php');

class Car {
    public  $id;
    public  $license;
    public  $driver;
    protected $passenger;

    public function __construct($license, Account $driver)
    {
        $this->license  = $license;
        $this->driver   = $driver;       
    }

    public function printDataCar(){
        if($this->passenger != null){
        echo "
        License : $this->license 
        Conductor: {$this->driver->name}
        Document: {$this->driver->document} 
        Passengers: $this->passenger
        
        ";
        }
    }

    public function getPassenger(){
        return $this->passenger;
    }

    public function setPassenger($passenger){

        if($passenger == 4){
            $this->passenger = $passenger;
        }else{
            echo " Necesitas asignar 4 pasajeros";
        }
    }
}

File: Java/Car.java
class Car {
    private Integer id;
    private String license;
    private Account driver;
    protected Integer passenger; //encapsulamiento
    
    public Car(String license, Account driver){
        this.license = license; //this.license hace referencia a la variable global y license es la variable local.
        this.driver = driver;
    }

    void printDataCar(){
        if(passenger != null){
        System.out.println("License: " + license + " Driver: " + driver.name + " Passenger : " + passenger);
        }
    }

    public Integer getPassenger(){
        return passenger;
    }
    public void setPassenger(Integer passenger){
        if(passenger == 4){
            this.passenger = passenger;
        }else{
            System.out.println("Necesitas asignar 4 pasajeros");
        }
        
    }

    public Integer getId() {
        return id;
    }

    public String getLicense() {
        return license;
    }

    public Account getDriver() {
        return driver;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public void setLicense(String license) {
        this.license = license;
    }

    public void setDriver(Account driver) {
        this.driver = driver;
    }
    
    
}